client/src/app/pages/waves/waves.tsx

```diff
@@ -0,0 +1,330 @@
+    } else {
+      delete newExpandedCells[application.name];
+    }
+    setExpandedCells(newExpandedCells);
+  };
+  const compoundExpandParams = (
+    application: Application,
+    columnKey: ColumnKey,
+    rowIndex: number,
+    columnIndex: number
+  ): TdProps["compoundExpand"] => ({
+    isExpanded: expandedCells[application.name] === columnKey,
+    onToggle: () =>
+      setCellExpanded(
+        application,
+        columnKey,
+        expandedCells[application.name] !== columnKey
+      ),
+    expandId: "composable-compound-expandable-example",
+    rowIndex,
+    columnIndex,
+  });
+  const handleIsRowSelected = (row: IComposableRow) => {
+    const matchingWave = waves.find((wave) => wave.name === row.name);
+    return matchingWave ? isRowSelected(matchingWave) : false;
+  };
+  const handleToggleRowSelected = (
+    row: IComposableRow,
+    isSelecting: boolean
+  ) => {
+    const matchingWave = waves.find((wave) => wave.name === row.name);
+    matchingWave && toggleRowSelected(matchingWave, isSelecting);
+  };
+  //RBAC
+  // xxxxWriteAccess = checkAccess(userScopes, waveWriteScopes);
+  const waveRowDropdownItems = true //TODO: Check RBAC access
+    ? [
+        <DropdownItem
+          key="bulk-export-to-issue-manager"
+          component="button"
+          // onClick={() => setExportIssueModalOpen(true)}
+        >
+          {t("actions.export")}
+        </DropdownItem>,
+        <DropdownItem
+          key="bulk-delete"
+          // onClick={() => {
+          // }}
+        >
+          {t("actions.delete")}
+        </DropdownItem>,
+      ]
+    : [];
+  //
+
+  const rows: IComposableRow[] = [];
+  currentPageItems?.forEach((item, index) => {
+    const expandedCellKey = expandedCells[item.name];
+    const isRowExpanded = !!expandedCellKey;
+    const isSelected = isRowSelected(item);
+    rows.push({
+      name: item.name,
+      isExpanded: isRowExpanded,
+      expandedCellKey: expandedCellKey,
+      isSelected: isSelected,
+      expandedContentMap: {
+        applications: (
+          <WaveApplicationsTable
+            applications={item.applications}
+          ></WaveApplicationsTable>
+        ),
+        stakeholders: (
+          <WaveStakeholdersTable
+            stakeholders={item.stakeholders}
+          ></WaveStakeholdersTable>
+        ),
+      },
+      cells: [
+        {
+          title: item.name,
+          width: 25,
+        },
+        {
+          title: dayjs(item.startDate).format("DD/MM/YYYY"),
+          width: 10,
+        },
+        {
+          title: dayjs(item.endDate).format("DD/MM/YYYY"),
+          width: 10,
+        },
+        {
+          title: item?.applications?.length.toString(),
+          compoundExpand: compoundExpandParams(item, "applications", index, 1),
+          width: 10,
+        },
+        {
+          title: item?.stakeholders?.length.toString(),
+          compoundExpand: compoundExpandParams(item, "stakeholders", index, 1),
```

Comment:
```
export interface TdCompoundExpandType {
  /** determines if the corresponding expansion row is open */
  isExpanded: boolean;
  /** Callback on toggling of the expansion */
  onToggle?: OnExpand;
  /** Id prefix for expandable cells **/
  expandId?: string;
  /** The row index */
  rowIndex?: number;
  /** The column index */
  columnIndex?: number;
}
```

Here we are just wrapping the toggle handler & other expand state specific data retrieval into a helper function that packages everything up for patternfly.